diff --git a/drivers/nxp/pn6xT/pn54x-i2c/pn54x.c b/drivers/nxp/pn6xT/pn54x-i2c/pn54x.c
index 805eb86..76c6085 100644
--- a/drivers/nxp/pn6xT/pn54x-i2c/pn54x.c
+++ b/drivers/nxp/pn6xT/pn54x-i2c/pn54x.c
@@ -62,7 +62,7 @@
 #include <linux/signal.h>
 #include "pn54x.h"
 
-#define NEXUS5x    0
+#define NEXUS5x    1
 #define DRAGON_NFC 1
 #define SIG_NFC 44
 #define MAX_BUFFER_SIZE 512
diff --git a/external/libese-spi/Android.mk b/external/libese-spi/Android.mk
index 2253a69..ef5142a 100644
--- a/external/libese-spi/Android.mk
+++ b/external/libese-spi/Android.mk
@@ -20,7 +20,7 @@ P73 := 2
 LOCAL_CFLAGS += -DP61=$(P61)
 LOCAL_CFLAGS += -DP73=$(P73)
 ## NXP_ESE_CHIP_TYPE := P61 to select  ESE P61 chip,  NXP_ESE_CHIP_TYPE := P73 to selectESE  P73 chip ###
-NXP_ESE_CHIP_TYPE := $(P73)
+NXP_ESE_CHIP_TYPE := $(P61)
 LOCAL_CFLAGS += -DNXP_ESE_CHIP_TYPE=$(NXP_ESE_CHIP_TYPE)
 ifeq ($(NXP_ESE_CHIP_TYPE),$(P61))
 MAKEFILE_PATH :=   $(LOCAL_PATH)/p61
diff --git a/external/libese-spi/p61/Android.mk b/external/libese-spi/p61/Android.mk
index 595ae95..c6a5ef9 100644
--- a/external/libese-spi/p61/Android.mk
+++ b/external/libese-spi/p61/Android.mk
@@ -40,7 +40,7 @@ LOCAL_SRC_FILES := -Wall $(call all-c-files-under, .)  $(call all-cpp-files-unde
 
 ANDROID_VER := $(subst ., , $(PLATFORM_VERSION))
 ANDROID_VER := $(word 1, $(ANDROID_VER))
-LOCAL_ANDROID_M := FALSE
+LOCAL_ANDROID_M := TRUE
 ifeq ($(ANDROID_VER), 6)
 LOCAL_ANDROID_M := TRUE
 endif
diff --git a/external/libese-spi/p61/spm/phNxpEseP61_Spm.c b/external/libese-spi/p61/spm/phNxpEseP61_Spm.c
index c1fc204..2d38d89 100644
--- a/external/libese-spi/p61/spm/phNxpEseP61_Spm.c
+++ b/external/libese-spi/p61/spm/phNxpEseP61_Spm.c
@@ -20,7 +20,7 @@
 #include <sys/ioctl.h>
 #include <fcntl.h>
 #include <errno.h>
-#include <linux/p61.h>
+#include "p61.h"
 #include "phNxpEseP61_Spm.h"
 #include <phNxpEseHal.h>
 #include <phNxpEseProtocol.h>
diff --git a/external/libese-spi/p73/Android.mk b/external/libese-spi/p73/Android.mk
index 89539d1..f613a3b 100644
--- a/external/libese-spi/p73/Android.mk
+++ b/external/libese-spi/p73/Android.mk
@@ -61,7 +61,7 @@ ANDROID_VER := $(subst ., , $(PLATFORM_VERSION))
 ANDROID_VER := $(word 1, $(ANDROID_VER))
 LOCAL_ANDROID_M := FALSE
 ifeq ($(ANDROID_VER), 6)
-LOCAL_ANDROID_M := TRUE
+LOCAL_ANDROID_M := FALSE
 endif
 
 ifeq ($(LOCAL_ANDROID_M),FALSE)
