diff --git a/external/libese-spi/p73/Android.mk b/external/libese-spi/p73/Android.mk
index 89539d1..f4c7a90 100644
--- a/external/libese-spi/p73/Android.mk
+++ b/external/libese-spi/p73/Android.mk
@@ -59,7 +59,7 @@ LOCAL_SRC_FILES += \
 
 ANDROID_VER := $(subst ., , $(PLATFORM_VERSION))
 ANDROID_VER := $(word 1, $(ANDROID_VER))
-LOCAL_ANDROID_M := FALSE
+LOCAL_ANDROID_M := TRUE
 ifeq ($(ANDROID_VER), 6)
 LOCAL_ANDROID_M := TRUE
 endif
